receiver/otlpreceiver/otlphttp.go

```diff
@@ -159,11 +214,23 @@ func errorHandler(w http.ResponseWriter, r *http.Request, errMsg string, statusC
 // Pre-computed status with code=Internal to be used in case of a marshaling error.
 var fallbackMsg = []byte(`{"code": 13, "message": "failed to marshal error message"}`)
 
+const fallbackGrpcMessage = "failed to marshal error message"
+
 const fallbackContentType = "application/json"
 
 func writeResponse(w http.ResponseWriter, contentType string, statusCode int, rsp proto.Message) {
 	var err error
 	var msg []byte
+	if contentType == grpcContentType {
+		if s, ok := rsp.(*spb.Status); ok {
+			statusCode = int(s.Code)
+			w.Header().Set(grpcMessage, s.Message)
+			rsp = emptyMessage
+		} else {
+			statusCode = int(codes.OK)
+		}
```

Comment:
Any links to gRPC specs that make it clear that this is the right thing to do? Similarly for other code that deals with gRPC peculiarities.